framework/auth/core.py

```diff
@@ -709,8 +712,6 @@ def add_unconfirmed_email(self, email, expiration=None):
 
     def remove_unconfirmed_email(self, email):
         """Remove an unconfirmed email addresses and their tokens."""
-        if email == self.username:
```

Comment:
When this function is called, your primary email should always be verified which will never be in the unconfirmed email list. If it is, then it is wrong with other code. But this check is not necessary.